### PR TITLE
[SIL-86] Rename user and auth network classes

### PR DESCRIPTION
**Pull request by kezc**
This is my proposal for splitting network parts, however, I'm not so sure if it's a good approach.
The problem raises from the fact, that auth requests require a different service than the rest of the calls, as it has a different API URL. Due to that, I couldn't place `getUser` inside `UserService`, so I decided to rename it to `AuthService` and create a new `GithubUserService`. As a chain of events, this resulted in renaming data sources and repositories too.

In conclusion:
GithubApi -> GithubAuthService
UserRepository -> AuthRepository
UserDataSource -> AuthDataSource 

new -> UserDataSource
new -> GithubUserService

### CODE REVIEW

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt [hunk 1]
```diff
@@ -22,25 +23,31 @@ object GithubModule {
 
     @Singleton
     @Provides
-    fun provideGithubApi(@AuthAPI retrofit: Retrofit): GithubApi =
-        retrofit.create(GithubApi::class.java)
+    fun provideGithubAuthService(@AuthAPI retrofit: Retrofit): GithubAuthService =
+        retrofit.create(GithubAuthService::class.java)
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: **Suggestion:** Removing the prefix `Github` from everywhere. 
I am not sure now if this prefix gives us additional value. There is no other API then GitHub API.  
What do you think? Do you consider it helpful? 

I mean to rename components: 
`GithubAuthService` -> `AuthService`
`GithubUserService` -> `UserService`
`GithubPullRequestsService` -> `PullRequestsService' 
etc..

File: app/src/main/java/com/appunite/loudius/network/datasource/UserDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.appunite.loudius.network.model.User
+import com.appunite.loudius.network.services.GithubUserService
+import com.appunite.loudius.network.utils.safeApiCall
+import javax.inject.Inject
+
+class UserDataSource @Inject constructor(private val userService: GithubUserService) {
+    suspend fun getUser(): Result<User> = safeApiCall {
+        userService.getUser()
+    }
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/network/datasource/UserDataSource.kt]: **Question: ** Tests in the separate PR right? 😅

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt [hunk 3]
```diff
@@ -22,25 +23,31 @@ object GithubModule {
...
+    ): AuthRepository = AuthRepositoryImpl(authDataSource, userLocalDataSource)
 
     @Singleton
     @Provides
-    fun provideUserDataSource(
-        api: GithubApi,
-    ): UserDataSource = UserNetworkDataSource(api)
+    fun provideAuthServiceDataSource(
+        service: GithubAuthService,
+    ): AuthDataSource = AuthNetworkDataSource(service)
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: **Steve Wonder:** I'm wondering if we want in same module dependencies for services and repos/data sources.
What opinions do you have? 🤔

**Review by Krzysiudan — APPROVED** [review]
Lgtm. That's cleaner for sure. 👌

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 1 above)
Comment (nowakweronika) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: Both forms are ok to me :)

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 3 above)
Comment (nowakweronika) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: It's ok to me, I don't like too much modules in small project, I don't see benefits from it.

File: app/src/main/java/com/appunite/loudius/network/datasource/UserDataSource.kt (hunk 2 above)
Comment (kezc) [reply app/src/main/java/com/appunite/loudius/network/datasource/UserDataSource.kt]: of course 😆

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 3 above)
Comment (kezc) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: maybe we could separate it by layer, but I don't have a strong opinion on that. Maybe I could add a ticket so we can think about it before showing publicly our app or if someone will have nothing to do?

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 1 above)
Comment (kezc) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: I fully agree, it annoys me :P I have added these changes to this PR, however, I have left GithubModule as it is.

**Review by nowakweronika — APPROVED** [review]
lgtm

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 1 above)
Comment (Krzysiudan) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: Nice 👌

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 3 above)
Comment (Krzysiudan) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: Creating a ticket for that sounds reasonable 👌
